src/torchcodec/_core/CudaDeviceInterface.cpp

```diff
@@ -265,37 +303,37 @@ void CudaDeviceInterface::convertAVFrameToFrameOutput(
     dst = allocateEmptyHWCTensor(height, width, device_);
   }
 
-  // Use the user-requested GPU for running the NPP kernel.
-  c10::cuda::CUDAGuard deviceGuard(device_);
```

Comment:
This guard isn't needed anymore as we now explicitly pass the current device to the NppContext creation.